caikit/runtime/http_server.py

```diff
@@ -0,0 +1,460 @@
+from caikit.core.toolkit.sync_to_async import async_wrap_iter
+from caikit.runtime.server_base import RuntimeServerBase
+from caikit.runtime.service_factory import ServicePackage, ServicePackageFactory
+from caikit.runtime.service_generation.rpcs import CaikitRPCBase
+from caikit.runtime.servicers.global_predict_servicer import GlobalPredictServicer
+from caikit.runtime.types.caikit_runtime_exception import CaikitRuntimeException
+import caikit.core.toolkit.logging
+
+## Globals #####################################################################
+
+log = alog.use_channel("HTTP")
+
+# Registry of DM -> Pydantic model mapping to avoid errors when reusing messages
+# across endpoings
+PYDANTIC_REGISTRY = {}
+
+
+# Mapping from GRPC codes to their corresponding HTTP codes
+# pylint: disable=line-too-long
+# CITE: https://chromium.googlesource.com/external/github.com/grpc/grpc/+/refs/tags/v1.21.4-pre1/doc/statuscodes.md
+GRPC_CODE_TO_HTTP = {
+    StatusCode.OK: 200,
+    StatusCode.INVALID_ARGUMENT: 400,
+    StatusCode.FAILED_PRECONDITION: 400,
+    StatusCode.OUT_OF_RANGE: 400,
+    StatusCode.UNAUTHENTICATED: 401,
+    StatusCode.PERMISSION_DENIED: 403,
+    StatusCode.NOT_FOUND: 404,
+    StatusCode.ALREADY_EXISTS: 409,
+    StatusCode.ABORTED: 409,
+    StatusCode.RESOURCE_EXHAUSTED: 429,
+    StatusCode.CANCELLED: 499,
+    StatusCode.UNKNOWN: 500,
+    StatusCode.DATA_LOSS: 500,
+    StatusCode.UNIMPLEMENTED: 501,
+    StatusCode.UNAVAILABLE: 501,
+    StatusCode.DEADLINE_EXCEEDED: 504,
+}
+
+
+# These keys are used to define the logical sections of the request and response
+# data structures.
+REQUIRED_INPUTS_KEY = "inputs"
+OPTIONAL_INPUTS_KEY = "parameters"
+
+## RuntimeHTTPServer ###########################################################
+
+
+class RuntimeHTTPServer(RuntimeServerBase):
+    """An implementation of a FastAPI server that serves caikit runtimes"""
+
+    ###############
+    ## Interface ##
+    ###############
+
+    def __init__(self, tls_config_override: Optional[aconfig.Config] = None):
+        super().__init__(get_config().runtime.http.port, tls_config_override)
+        # if not set in config, timeout is None and unvicorn accepts None or number of seconds
+        unvicorn_timeout_graceful_shutdown = get_config().runtime.http.timeout
+
+        self.app = FastAPI()
+
+        # Set up the central predict servicer
+        inference_service = ServicePackageFactory().get_service_package(
+            ServicePackageFactory.ServiceType.INFERENCE,
+        )
+        self.global_predict_servicer = GlobalPredictServicer(inference_service)
+
+        # Set up the central train servicer
+        # TODO: uncomment later on
+        # train_service = ServicePackageFactory().get_service_package(
+        #     ServicePackageFactory.ServiceType.TRAINING,
+        # )
+        # self.global_train_servicer = GlobalTrainServicer(train_service)
+        self.package_name = inference_service.descriptor.full_name.rsplit(".", 1)[0]
```

Comment:
NIT: Can we throw a newline and comment between the commented train stuff and the package name? A bit confusing as written since it makes it look like the package name is there instead of the train service (which it's not)